pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -212,6 +212,7 @@ def save_checkpoint(self, trainer, pl_module):
         ):
             return
 
+        trainer.accelerator.barrier()
```

Comment:
Nice, can you just add a comment to explain why this is done?
```suggestion
        # Synchronize all processes if using distributed
        trainer.accelerator.barrier()
```